qiime2/core/archive/tests/test_provenance.py

```diff
@@ -249,6 +251,17 @@ def test_unioned_primitives(self):
         self.assertIn('foo: 3', prov_yml)
         self.assertIn('bar: 2', prov_yml)
 
+    @mock.patch('qiime2.core.archive.provenance.tzlocal.get_localzone',
+                side_effect=ValueError())
+    def test_ts_to_date(self, mocked_link):
+        q2_paper_date = 1563984000
+
+        obs = str(provenance._ts_to_date(q2_paper_date))
+        exp = "2019-07-24 09:00:00"
+
+        self.assertEqual(obs, exp)
+        assert mocked_link.called
```

Comment:
Regarding the test failure on travis - I think you should move this assertion up to _before_ checking that obs == exp, that way you can verify that the patch is being applied correctly. Put another way, we should make sure that the code path [here](https://github.com/qiime2/qiime2/pull/513/files#diff-e183f9c47f8eeab74f2219b35f12921cR34) is actually being exercised in this test.

Reply:
Okay scratch that, I think I know what is wrong. Check out the docs for [`datetime.fromtimestamp`](https://docs.python.org/3.6/library/datetime.html#datetime.datetime.fromtimestamp), in particular, this part:

> If optional argument tz is None or not specified, the timestamp is converted to the platform’s local date and time, and the returned datetime object is naive.

So here is what I think is happening: when [this bit](https://github.com/qiime2/qiime2/pull/513/files#diff-e183f9c47f8eeab74f2219b35f12921cR35) of code is being executed, the `None` value signals to `datetime` to check for the system's timezone info. Since the travis env apparently _does_ have timezone info, the date/time is altered to match the system timezone which isn't really what we want.

I think a more robust solution would be to explicitly set the fallback timezone to 'UTC', rather than `None`, I will leave that exercise (and research) up to you. Let me know if you need a hand!